tools/src/main/java/com/nedap/archie/flattener/CAttributeFlattener.java

```diff
@@ -266,21 +257,25 @@ private boolean shouldRemoveParent(CObject specializedChildCObject, CObject matc
             }
 
         }
-        //the last matching child should possibly replace the parent, the rest should just add
-        //if there is just one child, that's fine, it should still work
-        if(allMatchingChildren.get(allMatchingChildren.size()-1).getNodeId().equalsIgnoreCase(specializedChildCObject.getNodeId())) {
-            return shouldReplaceParent(matchingParentObject, allMatchingChildren);
+        boolean hasSameNodeIdInMatchingChildren = allMatchingChildren.stream().anyMatch(c -> c.getNodeId().equals(matchingParentObject.getNodeId()));
+        if(hasSameNodeIdInMatchingChildren) {
+            //if parent contains id2, and child as well, replace the exact same node with the exact child. Otherwise,
+            //add children and replace the last child.
+            if(specializedChildCObject.getNodeId().equalsIgnoreCase(matchingParentObject.getNodeId())) {
+                return true;//shouldReplaceParent(matchingParentObject, allMatchingChildren);
```

Comment:
I think this comment is old code, it should probably be removed.

This inner if statement can then be replace by a single return statement.

Reply:
you're right, I'll fix that